pkg/sources/s3/s3.go

```diff
@@ -302,14 +284,27 @@ func (s *Source) pageChunker(ctx context.Context, client *s3.S3, chunksChan chan
 				},
 				Verify: s.verify,
 			}
+			if handlers.HandleFile(reader, chunkSkel, chunksChan) {
+				return
+			}
+
+			if err := reader.Reset(); err != nil {
+				log.WithError(err).Error("Error resetting reader to start.")
+			}
+			reader.Stop()
+
+			for chunkData := range common.ChunkReader(reader) {
+				chunk := *chunkSkel
```

Comment:
I'm not certain here, but wouldn't this create a shallow copy? `SourceMetadata` is still a pointer so every `chunk` in this loop will point to the same `SourceMetadata` object.

Maybe a closure would be better:

```go
makeChunk := func(chunkData IdkTheType) *sources.Chunk {
    return &sources.Chunk{ /* ... */ }
}
```

Reply:
It does, but all of chunks created will be the same except for `Data`.

Reply:
I think that'll be okay as long as we're only ever reading the chunk SourceMetadata.